avocado/core/nrunner.py

```diff
@@ -36,6 +36,19 @@ def __repr__(self):
                           self.args, self.kwargs)
 
 
+def runnable_to_command_line_args(runnable):
+    args = ['-k', runnable.kind]
+    if runnable.uri is not None:
+        args.append('-u')
+        args.append(runnable.uri)
+
```

Comment:
If uri is empty, error is propagated.

example of existing uri:
avocado runnable-run -k exec-test -u /bin/sleep -a 1
{'status': 'running'}
{'status': 'running'}
{'status': 'pass', 'returncode': 0, 'stdout': b'', 'stderr': b''}

example of missing uri:
avocado runnable-run -k exec-test -a 2
expected str, bytes or os.PathLike object, not NoneType

If the problem should be handled previously, go ahead and merge the code as we will be safe at this point. Otherwise, we need to proper handle missing parameters.

Reply:
That's a good point... A similar situation can happen with an empty uri, that is:

```
python3 -m avocado.core.nrunner runnable-run -k exec -u ''
Traceback (most recent call last):
  File "/usr/lib64/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/cleber/src/avocado/avocado/avocado/core/nrunner.py", line 507, in <module>
    main()
  File "/home/cleber/src/avocado/avocado/avocado/core/nrunner.py", line 493, in main
    subcommand_runnable_run(args)
  File "/home/cleber/src/avocado/avocado/avocado/core/nrunner.py", line 210, in subcommand_runnable_run
    for status in runner.run():
  File "/home/cleber/src/avocado/avocado/avocado/core/nrunner.py", line 78, in run
    stderr=subprocess.PIPE)
  File "/usr/lib64/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/usr/lib64/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
PermissionError: [Errno 13] Permission denied: ''
```

One way to look at this is that the `-u URI` parameter should be mandatory on the command line.  But, for other runners even something like "this_is_an_uri" can cause a similar error.  I'm leaning towards *not* implementing a policy at this level at this moment, and leave it to the discretion of each runner implementation.  The "job" component, should (but still doesn't) catch runner failures and properly report them.  The runner though, should be concerned with the "runnable" failures.  Also, the runner is usually not something that the end user will run directly.

Anyway, I'll keep this AS IS for now, as we can always make the behavior more strict in the future.